diff --git a/kernel/conftest.sh b/kernel/conftest.sh
index 0f6d1fc1a7fd..2b6c19c4fba6 100755
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -2142,10 +2142,15 @@
             #
             CODE="
             $CONFTEST_PREAMBLE
+            #include <linux/version.h>
             #include <asm/kvm_host.h>
 
             bool is_enable_apicv_present() {
+            #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 18, 0)
+                #error 'The enable_apicv symbol is restricted to the KVM modules'
+            #else
                 return enable_apicv;
+            #endif
             }"
 
             compile_check_conftest "$CODE" "NV_ENABLE_APICV_PRESENT" "" "types"
